### PR TITLE
[FLINK-3781] BlobClient may be left unclosed in BlobCache#deleteGlobal()

### PR DESCRIPTION
**Pull request by gaoyike**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

---------------------------------------------------------------------------------------------------------------------

Sorry for previous wrong pull-request, i forget to rebase my git

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobCache.java [hunk 1]
```diff
@@ -200,10 +200,14 @@ public void delete(BlobKey key) throws IOException{
 	 * @param key referring to the file to be deleted
 	 */
 	public void deleteGlobal(BlobKey key) throws IOException {
-		delete(key);
 		BlobClient bc = createClient();
-		bc.delete(key);
-		bc.close();
+		delete(key);
```
Comment (uce) [inline flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobCache.java]: BlobCache.delete can throw IOException (although right now it doesn't) and then `bc` is not closed again. Maybe move it up again and then it should be fine.

**Comment by uce** [comment]
Thanks for the contribution! I had an inline comment. If that's fixed, this looks good.

**Comment by gaoyike** [comment]
Updated!

Thanks uce!

**Comment by zentol** [comment]
a random commit regarding VeryFacstReservoirSampler has been added to this PR, could you get rid of it?

**Comment by gaoyike** [comment]
Done!

**Comment by fhueske** [comment]
Merging this PR

**Pull request closed without merging**